### ISSUE TITLE
Option for polars.DataFrame.to_dicts to skip null fields entirely

### ISSUE THREAD

**Issue by Guillermogsjc** [pola-rs/polars#12795]
### Description

`polars.DataFrame.to_dicts` is done sometimes to deliver the data into documental databases (like transforming data to upload to MongoDB for example). In those documental schemas, is not idiomatic to write in the documents fields that have `null` or `nan` value, in order to save space.

Could have an option `to_dicts` that would drop those fields whose value are `null` or `nan` in the resulting dictionary, in an optimal manner? 

At pandas you can 

```
NAN_SENTINELS = ["NaN", "nan", "np.nan"]

class NotNanRootDict(dict):

    @staticmethod
    def is_nan(v):
        if isinstance(v, dict) or isinstance(v, list):
            val = False
        else:
            val = pd.isnull(v) or v in NAN_SENTINELS

        return val

    def __new__(self, a):
        return {k: v for k, v in a if not self.is_nan(v)}

data_dict_list = df.to_dict(orient='records', into=NotNanRootDict)
```

It would be ideal to have at polars this `df.to_dicts(drop_nulls=True)` option that drops any field with `null` value, even in nested types (specially on structs), in an optimized performant manner (as polars uses to reach).

**Comment by deanm0000** [comment]
Maybe call it `drop_null_fields` since `drop_nulls` is an existing method that drops rows which I don't think is what you want.

Also, just to confirm you'd want this behavior:

```
df=pl.DataFrame({'a':[None,2,3], 'b':[2, 3,None]})
df.to_dicts()
[{'b': 2}, {'a': 2, 'b': 3}, {'a': 3}]
```

Do I have that right?

**Comment by Guillermogsjc** [comment]
yes, something passing a test from a list of dictionaries with nested elements to polars then back to the same, but without **fields whose values are nulls or NaN**, specially from structs, behaviour on lists with null elements to be specified.

orientative (to be designed):

```python
def test_drop_null_fields():
    nested_dict = [{    
        "key2": "value2",
        "key3": {        
            "subkey2": "subvalue2",
            "subkey_3": "b"
        },
        "key4": [{
            "a": 3
        }, {
            "c": 8
        }, {
            "d": 5
        }]
    },
    {
        "key1": 5.0,    
        "key3": {        
            "subkey2": "subvalue5",
            "subkey_3": "b"
        },
        "key4": [{
            "a": 3
        }, {
            "c": 8
        }, {
            "d": 5
        }]
    }]
    # HONOR THIS ASSERT (WITHOUT ORDERING REALLY)
    assert pl.DataFrame(nested_dict).to_dicts(drop_null_fields=True) == nested_dict
```

**Comment by orlp** [comment]
@Guillermogsjc NaN values getting skipped/ignored is non-negotiable and will not happen, that meaningfully changes the output.

**Comment by orlp** [comment]
No, dropping nulls inside lists also changes the behavior.

**Comment by kalatsu** [comment]
Hi there, I'm looking for this exact feature as well.  Currently, I have to do this using a loop that checks for returned 'None' values and rebuilds the outputted list. Doing this is still faster than using the pandas dictionary/filter conversion but it would be great to have it as a mirrored functionality as well.  

I've also testing this loop using a cython compilation and used a PyO3/polars-rust function to directly reference the dataframe so that it could be done without the initial conversion cost but it's a lot of effort to jump through those hoops and build out the functions and I'm not well versed in Rust to have an optimal implementation for that either.

Any chance that this item could be revisited for those that want to alter their list/dictionary output?

Thanks!
